src/util/system.h

```diff
@@ -129,12 +129,23 @@ class ArgsManager
 {
 public:
     enum Flags {
-        NONE = 0x00,
-        // Boolean options can accept negation syntax -noOPTION or -noOPTION=1
-        ALLOW_BOOL = 0x01,
-        ALLOW_INT = 0x02,
-        ALLOW_STRING = 0x04,
-        ALLOW_ANY = ALLOW_BOOL | ALLOW_INT | ALLOW_STRING,
+        NONE = 0x0, //<! Indicates flag lookup failed.
+
+        /* Low level validation flags. In most cases these should be avoided,
+         * and standard TYPE_* flag combinations below should be preferred. */
+        ALLOW_ANY      = 0x01, //!< disable validation
+        ALLOW_NEGATED  = 0x02, //!< allow -nofoo and -nofoo=1
```

Comment:
Nice to see `ALLOW_NEGATED` flag [again](https://github.com/bitcoin/bitcoin/pull/16097#issuecomment-514627823) ;)